service/history/workflow/mutable_state_impl.go

```diff
@@ -5066,6 +5103,17 @@ func (ms *MutableStateImpl) AddContinueAsNewEvent(
 				RunId:      ms.executionInfo.RootRunId,
 			},
 		}
+
+		startEvent, err := ms.GetStartEvent(ctx)
+		if err != nil {
+			return nil, nil, err
+		}
+		parentVersioningInfo := startEvent.GetWorkflowExecutionStartedEventAttributes().GetParentVersioningInfo()
+		// only set parent versioning info task queue if different from new workflow task queue
+		if command.GetTaskQueue().GetName() == parentVersioningInfo.GetTaskQueue() {
+			parentVersioningInfo.TaskQueue = ""
+		}
```

Comment:
we should be careful to account for command.GetTaskQueue() being nil, also for the fact that maybe parent tq and current tq and next tq are 3 different tqs.

```suggestion
                 currentTq := ms.GetExecutionInfo().GetTaskQueue()
                 parentTq := parentVersioningInfo.GetTaskQueue()
                 if parentTq == "" {
                   parentTq = currentTq
                 }
                 nextTq = command.GetTaskQueue().GetName()
		if nextTq == "" {
                   nextTq = currentTq
                 }
		if nextTq == parentTq {
			parentVersioningInfo.TaskQueue = ""
		} else {
		        parentVersioningInfo.TaskQueue = parentTq
		}
```